### PR TITLE
Fix Parse Object types

### PR DESCRIPTION
**Pull request by yagotome**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/parse-community/Parse-SDK-JS/blob/master/src/ParseObject.js
- [ ] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by jlnquere — APPROVED** [review]
Sounds good to me 👍 . Thanks !

File: types/parse/index.d.ts [hunk 1]
```diff
@@ -343,9 +344,9 @@ declare namespace Parse {
         save(attrs?: { [key: string]: any } | null, options?: Object.SaveOptions): Promise<this>;
         save(key: string, value: any, options?: Object.SaveOptions): Promise<this>;
         save(attrs: object, options?: Object.SaveOptions): Promise<this>;
-        set(key: string, value: any, options?: Object.SetOptions): boolean;
-        set(attrs: object, options?: Object.SetOptions): boolean;
-        setACL(acl: ACL, options?: SuccessFailureOptions): boolean;
+        set(key: string, value: any, options?: Object.SetOptions): this | boolean;
+        set(attrs: object, options?: Object.SetOptions): this | boolean;
+        setACL(acl: ACL, options?: SuccessFailureOptions): this | boolean;
```
Comment (amcasey) [inline types/parse/index.d.ts]: The documented return type appears to be `boolean`: https://parseplatform.org/Parse-SDK-JS/api/master/Parse.Object.html#setACL

Maybe add a test showing an example of a non-boolean return?  (Guess: these calls are chainable.)

Comment (amcasey) [reply types/parse/index.d.ts]: Oops, I missed your link to the actual source.  Looks like it's actually `this | false`.

Comment (yagotome) [reply types/parse/index.d.ts]: @amcasey You're right! I'm going to change all occurrences to `this | false`.

**Review by amcasey — APPROVED** [review]

**Pull request merged**